docs/preparing_recipes_for_conan2.md

```diff
@@ -0,0 +1,368 @@
+        ...
+```
+
+### Understanding some workarounds with the .names attribute model in recipes
+
+The ``.names`` model has some limitationsBecause of this, there are some recurrent
+workarounds in recipes to achieve things like setting absolute names for targets (without
+the ``::`` namespace), or for setting a custom namespace. These workarounds can now be
+undone with the ``set_property`` model because it allows setting arbitrary names for CMake
+targets. Let's see some examples of these workarounds in recipes:
+
+* **Use of components to get arbitrary target names in recipes**. Some recipes add a component
+  whose only role is to get a target name that is not limited by the namespaces added by
+  the legacy generators automatically. For example, the [ktx
+  recipe](https://github.com/conan-io/conan-center-index/blob/5753f954027d9d04b6d05e326f2757ab6b1ac69c/recipes/ktx/all/conanfile.py)
+  uses this workaround to get a target with name ``KTX::ktx``.
+
+```python
+class KtxConan(ConanFile):
+    name = "ktx"
+    ...
+
+    def package_info(self):
+        # changes namespace to KTX::
+        self.cpp_info.names["cmake_find_package"] = "KTX" 
+        ...
+        # the target inherits the KTX:: namespace and sets the target KTX::ktx
+        self.cpp_info.components["libktx"].names["cmake_find_package"] = "ktx"
+        ...
+        # all the information is set via this "fake root" component
+        self.cpp_info.components["libktx"].libs = ["ktx"]
+        self.cpp_info.components["libktx"].defines = [
+            "KTX_FEATURE_KTX1", "KTX_FEATURE_KTX2", "KTX_FEATURE_WRITE"
+        ]
+        ...
+```
+
+To add this information with the new model, just add the desired root ``cpp_info`` target
+name with ``cmake_target_name``. The target will inherit all the properties from the "fake
+root" component, so there's no need to redefine those. Please note that when the migration
+to Conan 2.0 is done, there will be no need for that component anymore and it should
+dissapear.
+
+```python
+class KtxConan(ConanFile):
+    name = "ktx"
+    ...
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "KTX" 
+        ...
+        # FIXME: Remove the libktx component in Conan 2.0, this is just needed for
+        # compatibility with legacy generators
+        self.cpp_info.components["libktx"].names["cmake_find_package"] = "ktx"
+        ...
+        self.cpp_info.components["libktx"].libs = ["ktx"]
+        self.cpp_info.components["libktx"].defines = [
+            "KTX_FEATURE_KTX1", "KTX_FEATURE_KTX2", "KTX_FEATURE_WRITE"
+        ]
+
+        # Set the root cpp_info target name as KTX::ktx
+        # it will inherit all the libbrary properties (libs, defines, etc.) 
+        # from the defined component. In Conan 2.0 the component should be removed
+        # and those properties should be added to the root cpp_info instead
+        self.cpp_info.set_property("cmake_target_name", "KTX::ktx")
```

Comment:
What is the outcome of this recipe in `CMakeDeps` generator? Should we add the defines to the _root_ target to get not just the name, but the same content?

Reply:
In this case where it's just one component which only purpose is to modify the target name we can leave this this way and the root cpp_info will inherit all the defines, libs and so on from the component... In other cases we may also want to define the component in the new model.